File: 22/22.rb
total = 0
names = File.read("names.txt").split(",").map! do |x|
  x.tr_s! '"', ""
end.sort!.each_with_index do |x, i|
  total += x.split("").map! do |c|
    c.ord - 'A'.ord + 1
  end.reduce(:+) * (i + 1)
end

puts total

File: 22/22.clj
(require '[clojure.string :as str])

(defn alpha-value [name]
  (reduce + (map #(- (int %) (int \@)) name)))

(let [names (-> (slurp "names.txt")
                (str/replace #"\"" "")
                str/trim
                (str/split #","))]
  (reduce-kv (fn [sum idx name]
               (+ sum (* (inc idx) (alpha-value name))))
             0
             (vec (sort names))))
